src/create-plugin.js

```diff
@@ -2,10 +2,17 @@
 
 import type {FusionPlugin} from './types.js';
 
-export function createPlugin<Deps, Service>(
-  opts: $Exact<FusionPlugin<Deps, Service>>
-): FusionPlugin<Deps, Service> {
+// eslint-disable-next-line
+type FusionPluginNoHidden<TDeps, TService> = $Diff<
+  FusionPlugin<TDeps, TService>,
+  {__plugin__: boolean}
+>;
+
+export function createPlugin<TDeps, TService>(
+  opts: $Exact<FusionPluginNoHidden<TDeps, TService>>
+): FusionPlugin<TDeps, TService> {
   // $FlowFixMe
   opts.__plugin__ = true;
+  // $FlowFixMe
```

Comment:
Not quite clear to me why we're adding the FlowFixMe here, I wonder if we should add comments above FlowFixMe with the reason?

Reply:
Yeah, it would be nice if Flow was smarter here.  We do abuse the type system right above, which is problematic due to `opts` being a [sealed](https://flow.org/en/docs/types/objects/#toc-sealed-objects) object so we cannot add `__plugin__` to it which makes Flow complain first on line 14:

> Cannot assign true to opts.__plugin__ because property __plugin__ is missing in FusionPlugin [1].

Likely related?

Alternatively, we can slightly change this to no longer be destructive and remove both of these!

```
export function createPlugin<TDeps, TService>(
  opts: $Exact<FusionPluginNoHidden<TDeps, TService>>
): FusionPlugin<TDeps, TService> {
  return {
    __plugin__: true,
    ...opts,
  };
}
```

It may be problematic if folks aren't using the returned value.

Reply:
I like this new return object, seems like it probably wouldn't cause many problems?

Reply:
I'll make a separate PR for that since it could be argued it's a breaking change.

Reply:
#223